cypress/integration/dashboard/dashboard_filtering_spec.js

```diff
@@ -0,0 +1,315 @@
+      cy.addDashboardPanels('Filter Bytes Test', 'search', false)
+
+      // Add filter
+      cy.addDashboardFilter('bytes', 'is', '12345678')
+
+      cy.get('[data-test-subj="filter filter-enabled filter-key-bytes filter-value-12,345,678 filter-unpinned "]').should('be.visible')
+    })
+    it('filters on pie charts', () => {
+      // Check that none of the pie charts are occupied with data (show "No results found")
+      cy.checkElementDoesNotExist('svg > g > g.arcs > path.slice')
+    })
+
+    it('area, bar and heatmap charts filtered', () => {
+      // Check that none of the charts are filled with data
+      cy.checkElementDoesNotExist('svg > g > g.series')
+    })
+
+    it('data tables are filtered', () => {
+      // Check that none of the data tables are filled with data
+      cy.checkElementDoesNotExist('[data-test-subj="paginated-table-body"] [data-cell-content]')
+    })
+
+    it('goal and guages are filtered', () => {
+      // Goal label should be 0, gauge label should be 0%
+      cy.checkElementContainsValue('svg > g > g > text.chart-label', 2, /^0%?$/)
+    })
+
+    it('tsvb time series shows no data message', () => {
+      // The no data message should be visible
+      cy.checkElementExists('[data-test-subj="noTSVBDataMessage"]', 1)
+    })
+
+    it('metric value shows no data', () => {
+      // The metrics should show '-'
+      cy.checkElementContainsValue('.mtrVis__value', 1, /^ - $/)
+    })
+
+    it('tag cloud values are filtered', () => {
+      cy.checkElementComponentDoesNotExist('[data-test-subj="tagCloudVisualization"]', 'svg > g > text')
+    })
+
+    it('tsvb metric is filtered', () => {
+      cy.checkElementContainsValue('[data-test-subj="tsvbMetricValue"]', 1, /^0 custom template$/)
+    })
+
+    it('tsvb top n is filtered', () => {
+      cy.checkElementContainsValue('[data-test-subj="tsvbTopNValue"]', 2, /^0$/)
+    })
+
+    it('saved search is filtered', () => {
+      cy.checkElementDoesNotExist('[data-test-subj="docTableExpandToggleColumn"]')
+    })
+
+    // TODO: Uncomment once https://github.com/elastic/kibana/issues/22561 is fixed
+    // it('timeline is filtered', async () => {
+    //   await dashboardExpect.timelineLegendCount(0)
+    // })
+
+    it('vega is filtered', () => {
+      cy.get('.vgaVis__view text').each(($el, index, $list) => {
+        cy.get($el).should('not.contain', /^5,000$/)
+      })
+    })
+  })
+
+  describe('using a pinned filter that excludes all data', () => {
+    before(() => {
+      cy.get('[data-test-subj="filter filter-enabled filter-key-bytes filter-value-12,345,678 filter-unpinned "]').click()
+      cy.get('[data-test-subj="pinFilter"]').click()
+    })
+
+    after(() => {
+      cy.get('[data-test-subj="filter filter-enabled filter-key-bytes filter-value-12,345,678 filter-pinned "]').click()
+      cy.get('[data-test-subj="pinFilter"]').click()
+    })
+
+    it('filters on pie charts', () => {
```

Comment:
Is this a repeat test? If not, maybe we should diff the test case name? For example, this one could be `it('pinned filter filters on pie charts'` . Some tests below have similar issue. I think it is better not to use the same test case name which will avoid confusion.

Reply:
I agree that it would be better to not use the exact same test case names across test suites, I changed the test case names to identify which test suite each individual test belonged to.